llvm/test/Transforms/SCCP/float-denormal-simplification.ll

```diff
@@ -0,0 +1,21 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+; RUN: opt -passes=sccp -S %s | FileCheck %s
+
+define float @test_ieee() #0 {
+; CHECK-LABEL: @test_ieee(
+; CHECK-NEXT:    ret float 0x36F4000000000000
+;
+  %1 = fmul float 2.802596928649634e-44, 2.000000e+00
```

Comment:
I would prefer a more meaningful test. We probably shouldn't be making constant folding dependent on the expected FP environment of the context function, and never flush them. This test should at minimum change to use a canonicalize intrinsic

Reply:
I mean passing the context should enable more optimizations than this. Replacing the fmul with a canonicalize should show the same thing, but be more true to the IR rules around where the flush would be guaranteed

Reply:
Understood.
I can work on adding tests that show other optimizations enabled by the context.
Replacing fmul with llvm.canonicalize doesn't reproduce the inconsistent behavior because the context was added to simplifyBinOp (canonicalize is handled elsewhere).
But, if you think the tests shouldn't look like this regardless, I'll work on changing them as well 👍

Reply:
> I mean passing the context should enable more optimizations than this

After diving more into the code it felt like the amount of optimizations is scarce for two reasons:
1. The binary operation that reaches SimplifyBinOp in this MR must have at least one operand that has the ValueLatticeElement tag: constant.
2. Integers has ValueLatticeElement tag: constantrange.

This leaves: floats and vector types as candidates for other optimizations that can appear in the test.

Regarding floats, I didn't find other optimizations that use the context passed to the function for other than deciding to flush the inputs/output. The context was used to find the denormal-fp-math attribute.

Regarding vector types, I don't have much experience in that area, but I couldn't write a test with a missed optimization opportunity in case the context is missing.

Therefore, with my currently limited knowledge of the code in general, I couldn't come up with better tests.